d2l-sort-by.html

```diff
@@ -0,0 +1,97 @@
+<link rel="import" href="../polymer/polymer-element.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown-menu.html">
+<link rel="import" href="../d2l-dropdown/d2l-dropdown-button-subtle.html">
+<link rel="import" href="../d2l-menu/d2l-menu.html">
+
+<dom-module id="d2l-sort-by">
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: inline-block;
+			}
+		</style>
+		<d2l-dropdown-button-subtle text="[[_buttonText]]">
+			<d2l-dropdown-menu no-padding no-auto-fit id="d2l-sort-by-menu">
+				<d2l-menu label="[[label]]">
+				  <slot></slot>
+				</d2l-menu>
+			  </d2l-dropdown-menu>
+		  </d2l-dropdown-button-subtle>
+	</template>
+	<script>
+		/**
+		 * `<d2l-sort-by>`
+		 * Polymer-based web component for D2L sort by component
+		 */
+		class SortBy extends Polymer.Element {
+			static get is() { return 'd2l-sort-by'; }
+			static get properties() {
+				return {
+					/**
+					* Reference to bound listener functions to be removed in disconnectedCallback
+					*/
+					_boundListeners: {
+						type: Object,
+						value: function() {
+							return {
+								_onItemSelect: null,
+							};
+						}
+					},
+					/**
+					* Text of the button
+					*/
+					_buttonText: {
+						type: String,
+						computed: 'getButtonText(selected)'
+					},
+					/**
+					* The currently selected option
+					*/
+					selected: {
+						type: String,
+						value: 'None', // TODO: langterm for default(none) selection
+						reflectToAttribute: true,
+					},
+					/**
+					* Label for the dropdown-menu
+					*/
+					label: {
+						type: String,
+						value: ''
+					}
+				};
+			}
+
+			connectedCallback() {
+				super.connectedCallback();
+				this._boundListeners = {
+					_onItemSelect: this._onItemSelect.bind(this),
+				};
+				this.$['d2l-sort-by-menu'].addEventListener('d2l-menu-item-select',
+					this._boundListeners._onItemSelect);
+			}
+
+			disconnectedCallback() {
+				super.disconnectedCallback();
+				this.$['d2l-sort-by-menu'].removeEventListener('d2l-menu-item-select',
+					this._boundListeners._onItemSelect);
+			}
+
+			_onItemSelect(e) {
+				this.selected = e.target.text;
+				this.dispatchEvent(new CustomEvent(
+					'd2l-sort-by-change',
+					{ bubbles: true, composed: true, detail: { value: this.selected } }
+				));
+				this.$['d2l-sort-by-menu'].close();
```

Comment:
I'm surprised selecting an item in the menu doesn't automatically close it?

Reply:
I also expected the menu to close automatically - my guess is that there's something going on with the events passing through shadow-dom and not reaching the `d2l-sort-by`'s `d2l-menu`

I'll do some more testing and see if changing the `d2l-sort-by-option` to use `d2l-menu-item-radio` will have any effects

Reply:
Selecting an item in the menu SHOULD automatically close it for sorting.

Reply:
I think figured out why this is happening: 

`d2l-sort-by-option` is just a `<d2l-menu-item-radio>`, and in the [code](https://github.com/BrightspaceUI/menu/blob/master/d2l-menu-item-radio.html#L56), it queries the parent (expecting a `d2l-menu`) for all the `d2l-menu-item-radio` elements and changes their `selected` state 

However, when that code runs in this context, the parent is `shadowRoot` (i.e., this element). It seems I can't just wrap the `d2l-menu-item-radio` as it is currently

Reply:
I think the behavior for selectable item is intentional.  In the case of `d2l-menu-item-checkbox` elements, the expectation is that the user or consumer closes the dropdown because `d2l-dropdown-menu` expects that the user may make more than one change.  `d2l-menu-item-radio` shares the "selectable" menu item behavior, and it may be possible to optimize the close logic for that case.

For standard `d2l-menu-item` components, the [d2l-dropdown-menu](https://github.com/BrightspaceUI/dropdown/blob/master/d2l-dropdown-menu.html#L96) behavior listens for the `d2l-menu-item-select` event and closes the menu if the item is a `d2l-menu-item` because the expectation is that the user will typically only perform one action.

Reply:
In terms of wrapping the menu items, I think that may not work well in a variety of ways, for a bunch of reasons.  The `d2l-menu` expects menu items to be inside.  If it's necessary to wrap them, then a custom menu item should be used, but I do not advise creating custom menu items.  Rather, I suggest using the built-in ones.